numerics/root_finders_body.hpp

```diff
@@ -164,8 +164,13 @@ absl::btree_set<Argument> DoubleBrent(Function f,
   absl::btree_set<Argument> zeroes_above;
   absl::btree_set<Argument> zeroes_below;
 
-  Argument a = lower_bound;
-  Argument b = upper_bound;
+  Argument const a = lower_bound;
+  Argument const b = upper_bound;
+
+  // These computations ensure a reasonable margin even if one of the bounds is
+  // zero.
+  Argument const a_effective = a + eps * std::max(Abs(a), Abs(b));
+  Argument const b_effective = b - eps * std::max(Abs(a), Abs(b));
```

Comment:
Surely this function should work for an affine `Argument`.

1. Add a test with a function Instant -> double;
2. Change this to use eps * Abs(a - b), or if that doesn’t work use subtractions of Argument{} and explain why it needs to be aware of the position of the origin.